index.js

```diff
@@ -159,13 +164,18 @@ module.exports = (opts = {}) => {
     },
 
     doUpdateIpfsApiAddress: (addr) => ({ store }) => {
-      addr = addr.split('/')
-
-      store.doUpdateIpfsApiOpts({
-        host: addr[2],
-        port: addr[4],
-        protocol: 'http'
-      })
+      try {
+        const maddr = multiaddr(addr).nodeAddress()
+        const https = addr.includes('https')
```

Comment:
addr.includes is easily fooled by dnsaddrs like `/dnsaddr/httpseverywhere/tcp/80`